pkg/services/provisioning/dashboards/types.go

```diff
@@ -91,13 +97,19 @@ func mapV0ToDashboardAsConfig(v0 []*DashboardsAsConfigV0) []*DashboardsAsConfig
 		})
 	}
 
-	return r
+	return r, nil
 }
 
-func (dc *DashboardAsConfigV1) mapToDashboardAsConfig() []*DashboardsAsConfig {
+func (dc *DashboardAsConfigV1) mapToDashboardAsConfig() ([]*DashboardsAsConfig, error) {
 	var r []*DashboardsAsConfig
+	seen := make(map[string]bool)
 
 	for _, v := range dc.Providers {
+		if _, ok := seen[v.Name.Value()]; ok {
+			return nil, fmt.Errorf("name %s is not unique", v.Name.Value())
```

Comment:
Not sure whether this should fail all configs when a duplicate is found and after or if we just should log an error? Leaning towards logging an error.

Reply:
@marefr See my answer above.